Lib/urllib/parse.py

```diff
@@ -500,7 +534,9 @@ def urlunsplit(components):
     empty query; the RFC states that these are equivalent)."""
     scheme, netloc, url, query, fragment, _coerce_result = (
                                           _coerce_args(*components))
-    if netloc or (scheme and scheme in uses_netloc and url[:2] != '//'):
+
+    scheme_classes = _scheme_classes(scheme)
+    if netloc or (scheme and (SchemeFlag.NETLOC & scheme_classes) and url[:2] != '//'):
```

Comment:
if netloc or NETLOC in scheme_classes and url[:2] != '//'):

if `scheme` is empty, then `scheme_classes` will be empty (`SchemeFlag(0)`) so we can just check if `NETLOC` is in `scheme_classes`.